core/app/models/spree/store_credit.rb

```diff
@@ -253,7 +253,7 @@ def validate_no_amount_used
     def associate_credit_type
       unless type_id
         credit_type_name = category.try(:non_expiring?) ? 'Non-expiring' : 'Expiring'
-        self.credit_type = Spree::StoreCreditType.find_by(name: credit_type_name)
+        self.credit_type = Spree::StoreCreditType.find_or_create_by(name: credit_type_name)
```

Comment:
@aleksandarpetrushev this is good, but we still need to create this in seeds - https://github.com/spree/spree/blob/main/core/app/services/spree/seeds/store_credit_categories.rb